lib/data_sources/holdings_db.rb

```diff
@@ -17,9 +17,9 @@ class HoldingsDB < SimpleDelegator
 
     def initialize
       @rawdb = self.class.connection
-      # Always check that we're actually connected and reconnect if necessary
+      # Check once every few seconds that we're actually connected and reconnect if necessary
       @rawdb.extension(:connection_validator)
-      @rawdb.pool.connection_validation_timeout = -1
+      @rawdb.pool.connection_validation_timeout = 5
```

Comment:
We can observe if we get any kind of connection drops -- I think it will make more sense to automatically retry than to constantly revalidate the connection.